hive-metastore/src/main/java/org/apache/iceberg/hive/HiveCatalog.java

```diff
@@ -707,7 +707,8 @@ protected String defaultWarehouseLocation(TableIdentifier tableIdentifier) {
           clients.run(client -> client.getDatabase(tableIdentifier.namespace().levels()[0]));
       if (databaseData.getLocationUri() != null) {
```

Comment:
You might want to change this to `Strings.isNullOrEmpty` otherwise you have a mismatch between this check and the one performed by `LocationUtil::stripTriailingSlash`